tex/chapters/Theory.tex

```diff
@@ -0,0 +1,142 @@
+\cfrac{y (1 - x)}{2y(1-x) + x(1-x) + y(1-y)}, \\
+\cfrac{y (1 - y)}{2y(1-x) + x(1-x) + y(1-y)}, \\
+\cfrac{y (1 - x)}{2y(1-x) + x(1-x) + y(1-y)}
+\end{bmatrix}
+\end{equation}
+\textbf{Step 4} - Calculate the expected score.
+
+\begin{equation}
+F = \pi \cdot
+\begin{bmatrix}
+3 \\
+0 \\
+5 \\
+1
+\end{bmatrix}
+=
+\cfrac{3x(1-x) + y(1-x) + 5y(1-y)}{2y(1-x) + x(1-x) + y(1-y)}
+\end{equation}
+
+\textbf{Step 5} - Plot the resulting function.
+
+\section{Finite State Machines}
+
+Figure \ref{fig:Tit4TatFSM} and figure \ref{fig:PavlovFSM} show the Finite State Machine (FSM) representations for Tit-For-Tat and Pavlov respectively.
+Nodes represent the previous action taken by the strategy and the opponent, ie node $(D, C)$ implies that is the preceding turn, the strategy chose to Defect and the opponent chose to co-operate.
+Arcs represent the choice made by the opponent at the current turn, and lead us to the state of the next turn.
+
+It should be noted that these figures are not necessarily the simplest representation of their corresponding strategy.
+For example, Tit-For-Tat requires no knowledge of its own previous moves, but they have been included for completeness.
+
+\begin{figure}[!hbtp]
+    \begin{center}
+        \includestandalone{../img/Tit4TatFSM}
+        \caption{FSM for TitForTat}\label{fig:Tit4TatFSM}
+        \includestandalone{../img/PavlovFSM}
+        \caption{FSM for Pavlov (Win-Stay Lose-Shift)}\label{fig:PavlovFSM}
+    \end{center}
+\end{figure}
+
+\begin{figure}[!hbtp]
+    \begin{center}
+        \includestandalone{../img/MajorityFSM}
+        \caption{FSM for Majority in a game with 4 Turns}\label{fig:MajorityFSM}
+    \end{center}
+\end{figure}
+
+In figure \ref{fig:MajorityFSM} we have a more complex FSM for the strategy Majority which plays in the following way:
+
+\begin{itemize}
+  \item If the opponent has cooperated the majority of the time, Majority will cooperate
+  \item If the opponent has defected the majority of the time, Majority will defect
+  \item Note - the strategy shown is technically Soft Majority, if the opponents cooperations and defections are equal it will cooperate. Hard Majority would defect in this situation.
+\end{itemize}
+
+This implies that the strategy Majority requires knowledge of all previous states, and therefore could not be represented as a FSM.
+However in Theorem \ref{thm:fsm} it is stated that if the number of turns in a game is known, any strategy can be represented as an FSM.
+This will be proved in two different ways:
+\begin{itemize}
+  \item Firstly, by extending the definition of an
+  \item If the opponent has defected the majority of the time, Majority will defect
+\end{itemize}
+A formal defintion of a Finite State Machine is given by Definition \ref{def:fsm} but first we will outline some motivating key characteristics of a system that can be modeled with a FSM:
+
+\begin{itemize}
+ \item The system must be describable by a finite set of states.
```

Comment:
I had missed the wording of the previous sentence:

> characteristics of a system

This is fine as it is. Leave it.